pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -92,6 +96,30 @@ type UpdateStrategy struct {
 	*RegistryPoll `json:"registryPoll,omitempty"`
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+// UnmarshalJSON checks to see the provided polling interval is valid. If so, use it, if not set it to the default interval.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) error {
+	var updateStrategy map[string]map[string]string
+	if err := json.Unmarshal(data, &updateStrategy); err != nil {
+		return err
+	}
+	if updateStrategy == nil {
+		return nil
+	}
+
+	registryPoll := &RegistryPoll{}
+	if rp, ok := updateStrategy["registryPoll"]; ok {
+		duration, err := time.ParseDuration(rp["interval"])
```

Comment:
Does this also require a nil check attempting to access a nested map key's value?

Reply:
I think in this case it would return the zero value which is not actually what we want. Good catch.